### PR TITLE
Fix language servers stop/start logic + fix memory leak

### PR DESCRIPTION
**Pull request by arnaudsjs**
changes:

* Fix bug that could cause multiple instances of the language server to be running simultaneously.
* Fix memory leak by resetting the `inmanta.plugins.PluginMeta` and `inmanta.export.Exporter` classes before every compile.

### CODE REVIEW

**Review by FloLey — APPROVED** [review]

File: src/language_server.ts [hunk 1]
```diff
@@ -619,33 +627,46 @@ export class LanguageServer {
...
+				await this._stopServerAndClient();
+			});
+		} else {
+			await this._stopServerAndClient();
+		}
+	}
+
+	/* 
+	 * Stops the language server and its client.
+	 * This function should always run under the `mutex` lock.
```
Comment (sanderr) [inline src/language_server.ts:655]: This says "always", yet `stopServerAndClient` has a boolean to control it.

**Review by sanderr — CHANGES_REQUESTED** [review]

File: src/language_server.ts (hunk 1 above)
Comment (arnaudsjs) [reply src/language_server.ts:655]: This docstring belongs to the `_stopServerAndClient()` method and not to the `stopServerAndClient` method

Comment (sanderr) [reply src/language_server.ts:655]: Yes, I meant to say that `stopServerAndClient`'s boolean controls whether or not it calls `_stopServerAndClient` under the mutex lock. So it violates the docstring of this method. Either the docstring is too strict, or the boolean should not be there in the caller method.

Comment (sanderr) [reply src/language_server.ts:655]: Or should the boolean be false only if it is already running under the lock?

Comment (arnaudsjs) [reply src/language_server.ts:655]: yes

**Review by sanderr — APPROVED** [review]

**Pull request merged**